app/models/manageiq/providers/ansible_tower/shared/automation_manager/job.rb

```diff
@@ -9,9 +9,10 @@ module ClassMethods
     #   :extra_vars => Hash
     #
     def create_stack(template, options = {})
-      self.new(:name                  => template.name,
+      template_ref = template.new_record? ? nil : template
+      new(:name                  => template.name,
           :ext_management_system => template.manager,
-          :job_template          => template).tap do |stack|
+          :job_template          => template_ref).tap do |stack|
```

Comment:
@bzwei - Setting this to nil isn't going to cause any other surprises?

It seems odd that we'll have a AnsibleTowerJob without a referenced JobTemplate (ConfigurationScript.) It almost feels like this should be a different type of subclassed Job since it will never have what a normal AnsibleTowerJob would have - a referenced JobTemplate.

In that case we could override create_stack specifically for the runner.

Reply:
So we don't have the use case that expects the template from a job (stack) can never be `nil`. For Azure provider it is actually common that we cannot get the template for a stack.

We could create a subclass of Job class to separate this use from normal Ansible jobs, but I feel it is not very helpful. When we query tower jobs the subclass instances are included anyway.